app/scripts/migrations/054.js

```diff
@@ -0,0 +1,57 @@
+import { cloneDeep } from 'lodash';
+
+const version = 54;
+
+/**
+ * Remove tokens that don't have an address due to
+ * lack of previous addToken validation.  Also removes
+ * an unwanted, undefined image property
+ */
+export default {
+  version,
+  async migrate(originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData);
+    versionedData.meta.version = version;
+
+    const { PreferencesController } = versionedData.data;
+
+    // These properties are arrays and can be directly filtered
+    const tokenArrays = ['tokens', 'hiddenTokens'];
+    tokenArrays.forEach((tokenKey) => {
+      if (PreferencesController[tokenKey]) {
+        PreferencesController[tokenKey] = PreferencesController[
+          tokenKey
+        ].filter(({ address }) => address);
+      }
+    });
+
+    // These properties are: (account) -> (network) -> [(token), ....]
+    const tokenObjects = ['accountHiddenTokens', 'accountTokens'];
+    tokenObjects.forEach((tokenObject) => {
+      if (PreferencesController[tokenObject]) {
+        Object.keys(PreferencesController[tokenObject]).forEach((account) => {
+          const networks = Object.keys(
+            PreferencesController[tokenObject][account],
+          );
+          networks.forEach((network) => {
+            PreferencesController[tokenObject][account][
+              network
+            ] = PreferencesController[tokenObject][account][network].filter(
+              ({ address }) => address,
```

Comment:
The values pointed to by network ids in the `accountHiddenTokens` objects are arrays of strings. They look like this:
```
    "accountHiddenTokens": {
      "0xd2eab055e00d7daeffec4ef62b65bd069e204704": {
        "0x1": [
          "0x2260fac5e5542a773aa44fbcfedf7c193bc2c599"
        ],
        "0x539": [],
        "0x3": []
      }
    },
```
So this filter would always result in an empty array when `tokenObject === 'accountHiddenTokens'`